### DISCUSSION TITLE
How can I control the "checked" after a button click?

### DISCUSSION THREAD

**Discussion by sonitsuka** [Craig-Creeger/react-likert-scale#19] [Q&A]
Hello and thank you for the great package! 

I have been stuck for about 2 weeks with the problem, "checked state".

    CHECKED key will PRE-check a radio button. 
    
    responses: [
      { value: 1, text: "Abysmal" },
      { value: 2, text: "Poor" },
      { value: 3, text: "Average", checked: true },
      { value: 4, text: "Good" },
      { value: 5, text: "Excellent" }
    ]

But how can I control this checked AFTER clicking the radio button?
I saw the console for the new state to check, but I couldn't find the "checked" in the array, Object and state. 

I'd like to use/call this likert Package for several times in a survey page, for example, 5 questions' survey & one question per page: I call the page 5 times.   
In addition, the chosen Answers are kept in the likert-scale each pages. 
Therefore, I need to know how can I control the checked Object.

**Comment by Craig-Creeger** [comment]
The "checked" attribute is used to set _**initial state on page load**_. Let's say someone selects a different radio button and you want to know which one was selected, you would use the `onChange` callback function. Something similar to this:

```
const onChange = (chosen) => {
    const userClickOn = `The user picked {chosen?.value}—{chosen?.text}`;
    console.log(userClickedOn);
};

<Likert onChange={onChange} question="foo" responses={[]} />
```

The `value` and `text` keys tell you what the user clicked on.

See this for a complete example:
https://codepen.io/craig-creeger/pen/WNGaJxe

**Comment by Craig-Creeger** [comment]
I think I understand your question now -- you are looking for a way to programatically change which radio button is selected. You will need to directly set it via Javascript, but since that is a React effect, the code should be inside a `useEffect` hook. Here's an example:

```
import Likert from "https://cdn.skypack.dev/react-likert-scale@latest";
const { useState, useRef, useEffect } = React;

const likertOptions = {
  question:
    "What is your opinion of the President of the United State’s performance?",
  responses: [
    { value: 1, text: "Abysmal" },
    { value: 2, text: "Poor" },
    { value: 3, text: "Average", checked: true },
    { value: 4, text: "Good" },
    { value: 5, text: "Excellent" }
  ]
};

function PreviewComponent() {
  const refLikert = useRef(null);
  const [chosen, setChosen] = useState(null);

  const onChange = (chosen) => {
    const content = (
      <span>
        You chose:{" "}
        <strong>
          {chosen?.value}—{chosen?.text}
        </strong>
      </span>
    );
    setChosen(content);
  };

  const currentPage = "page two";

  useEffect(() => {
    const secondRadio = refLikert.current.querySelectorAll(
      'input[type="radio"]'
    )[1];
    secondRadio.checked = true;
  }, [currentPage]);

  return (
    <>
      <Likert
        {...likertOptions}
        onChange={onChange}
        ref={refLikert}
        className="withBackground"
      />
      <p style={{ textAlign: "center" }}>{chosen}</p>
    </>
  );
}
```

**Reply by Craig-Creeger** [reply]
Also, this seems like a good feature to add to the component. I'll try to get that done within the next week or two.
